### PR TITLE
Emails via Sendgrid

### LINKED ISSUE #3336: Emails from dev environments
### Description
we currently have no way of receiving emails from dev

### Proposed solution 
Create and install sendgrid API keys for dev environments

### Additional context

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS PULL REQUEST

## Related Issue

- resolve #3336 

## Changes Proposed

- I've created a new SendGrid API key for our dev instances and updated the terraform to pull that key in.
- This enables dev as another deployed environment to send emails from.

## Additional Information

- If we want to enable additional envs permanently or temporarily, we should be able to do that just by changing the `sendgrid.enabled` header to `true`
- I didn't enable this in all environments because I have concerns about all the email addresses in our deployed environments. I'm worried that some match up with real emails accidentally. We should consider how we come up with our email addresses. I've used this pattern to create unique users and get them all in my inbox. `<my_real_email>+<unique_id>@domain.provider`
- Environments with SendGrid enabled are now `dev, pentest, and prod`

## Testing

- This is deployed to `dev`; create new tests and send emails.

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Review by zdeveloper — COMMENTED** [review]
LGTM! thank you for doing this. I think we should only enable it for some of the dev envs (maybe one or two) just to avoid sending real emails by accident and adding a banner on these envs that real emails will be sent.

**Review by zdeveloper — DISMISSED** [review]
Re-reviewing because I wrongfully assumed this was being turned on on all envs. Still think we should add some sort of warning on that env though.

**Comment by alismx** [comment]
> Re-reviewing because I wrongfully assumed this was being turned on on all envs. Still think we should add some sort of warning on that env though.

I think that's a great idea. I'll look into adding that banner along with cleaning up emails.

File: .github/actions/maintenance-banner-deploy/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Maintenance banner deploy
+description: Sends an email to the support team when the state of the maintenance banner changes in PROD
```
Comment (rin-skylight) [inline .github/actions/maintenance-banner-deploy/action.yml:2]: Is this description accurate? The email portion doesn't seem to be included here.

Comment (alismx) [reply .github/actions/maintenance-banner-deploy/action.yml:2]: It is not accurate! :fire:

Comment (alismx) [reply .github/actions/maintenance-banner-deploy/action.yml]: :heavy_check_mark:

**Review by zdeveloper — APPROVED** [review]
LGTM! thank you for adding the auto banner deployment

**Review by rin-skylight — APPROVED** [review]
🎉

**Pull request merged**